When PDFBox doesn't detects the number of pages the pdf annotation fails

When annotatePDF is invoked, the original pdf is opened with PDFBox. There are certain cases where PDFBox doesn't extract correctly the content (I haven't figured out why). 

However the call fails with 500 and the following error: 
```
[ERROR] org.grobid.service.process.GrobidRestProcessFiles: An unexpected exception occurs. 
java.lang.IndexOutOfBoundsException: Index: 20, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at org.grobid.core.visualization.CitationsVisualizer.annotatePage(CitationsVisualizer.java:131)
	at org.grobid.core.visualization.CitationsVisualizer.annotatePdfWithCitations(CitationsVisualizer.java:89)
	at org.grobid.service.process.GrobidRestProcessFiles.processPDFAnnotation(GrobidRestProcessFiles.java:626)
	at org.grobid.service.GrobidRestService.processAnnotatePDF(GrobidRestService.java:638)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.jersey.spi.container.JavaMethodInvokerFactory$1.invoke(JavaMethodInvokerFactory.java:60)
	at com.sun.jersey.server.impl.model.method.dispatch.AbstractResourceMethodDispatchProvider$ResponseOutInvoker._dispatch(AbstractResourceMethodDispatchProvider.java:205)
	at com.sun.jersey.server.impl.model.method.dispatch.ResourceJavaMethodDispatcher.dispatch(ResourceJavaMethodDispatcher.java:75)
	at com.sun.jersey.server.impl.uri.rules.HttpMethodRule.accept(HttpMethodRule.java:288)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.ResourceClassRule.accept(ResourceClassRule.java:108)
	at com.sun.jersey.server.impl.uri.rules.RightHandPathRule.accept(RightHandPathRule.java:147)
	at com.sun.jersey.server.impl.uri.rules.RootResourceClassesRule.accept(RootResourceClassesRule.java:84)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1469)
	at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1400)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1349)
	at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1339)
	at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:416)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:537)
	at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:699)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:487)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:362)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:181)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:726)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:405)
	at org.mortbay.jetty.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:206)
	at org.mortbay.jetty.handler.HandlerCollection.handle(HandlerCollection.java:114)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:324)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:505)
	at org.mortbay.jetty.HttpConnection$RequestHandler.content(HttpConnection.java:843)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:648)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:211)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:380)
	at org.mortbay.io.nio.SelectChannelEndPoint.run(SelectChannelEndPoint.java:395)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:488)
```
The DocumentContent of PDDocument is null, so in [CitationVisualizer](https://github.com/kermitt2/grobid/blob/master/grobid-core/src/main/java/org/grobid/core/visualization/CitationsVisualizer.java#L131) it's throwing an OutOfBoundException:

```
        PDPage page = (PDPage) document.getDocumentCatalog().getAllPages().get(pageNum.intValue());
```

My proposal: 
 1. for rare cases on which PDF box doesn't extract the number of pages (and the DocumentCatalog) - relateively easy to fix, but we should also fix #148 otherwise the user is not informed.
 2. update PDFBox? I can already say that version 2.0.3 is going to break quite some parts :) and I cannot really assure it would solve the problem. We can open a separate task in case we decide to do so.

--- a/grobid-core/src/main/java/org/grobid/core/visualization/CitationsVisualizer.java
+++ b/grobid-core/src/main/java/org/grobid/core/visualization/CitationsVisualizer.java
@@ -170,6 +170,16 @@ public class CitationsVisualizer {
         String[] split = coords.split(",");
 
         Long pageNum = Long.valueOf(split[0], 10) - 1;
+
+        if (pageNum < 0 || pageNum >= document.getNumberOfPages()) {
+            LOGGER.warn(
+                    "Annotation skipped: page index "
+                            + (pageNum + 1)
+                            + " is out of range (document reports "
+                            + document.getNumberOfPages()
+                            + " page(s)).");
+            return;
+        }
         PDPage page = document.getDocumentCatalog().getPages().get(pageNum.intValue());
 
         PDRectangle mediaBox = page.getCropBox();

